c/misra/src/rules/RULE-5-4/MacroIdentifiersNotDistinct.ql

```diff
@@ -0,0 +1,29 @@
+/**
+ * @id c/misra/macro-identifiers-not-distinct
+ * @name RULE-5-4: Macro identifiers shall be distinct
+ * @description Declaring multiple macros with the same name leads to undefined behaviour.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-5-4
+ *       correctness
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+from Macro m, Macro m2
+where
+  not isExcluded(m, Declarations1Package::macroIdentifiersNotDistinctQuery()) and
+  not m = m2 and
+  (
+    if m.getName().length() >= 64
+    then m.getName().substring(0, 62) = m2.getName().substring(0, 62)
+    else m.getName() = m2.getName()
```

Comment:
Can you document why this is 63 characters (e.g. due to C99), and note that we don't currently support C90's requirements here?

Reply:
is currently noted in [pkg json implementation note](https://github.com/knewbury01/codeql-coding-standards/blob/knewbury01/Declarations1/rule_packages/c/Declarations1.json#L120)s, should it be in both places? can inline in query also but dont want to be redundant

Reply:
The implementation notes are for developers reviewing results, whereas comments in the query are for ql writers. So I think it would be worth putting this information here, so it's easy to see when you read the query.

Reply:
done!